test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/UriParser/EnumFilterFunctionalTests.cs

```diff
@@ -94,11 +99,11 @@ public void ParseFilterWithEnum()
 
             binaryNode
                 .Left
-                .ShouldBeSingleValuePropertyAccessQueryNode(GetColorProp(this.userModel));
+                .ShouldBeSingleValuePropertyAccessQueryNode(this.GetIEdmProperty("Color"));

```

Comment:
Why are we changing these tests? Should we add new tests to show the new behavior, or the tests have to change?

Reply:
@ElizabethOkerio Thanks for bringing this up.

I only change the private methods for getting IEdmProperty and IEdmType from IEdmModel to make them reused across all the tests. I noted a lot of code repetition. For example, if I want to get Person property from the IEdmModel, I am forced to create `GetPersonProp` and the same way if I want to get Person type, I have to create `GetPersonType`.

I have created one method `private IEdmStructuralProperty GetIEdmProperty(string propertyName)` to replace the following methods so that in future you do not need to add another method to achieve the same thing. The only thing you need to do is to pass propertyName to `GetIEdmProperty()` and you are good to go.
```cs
private IEdmStructuralProperty GetColorProp(IEdmModel model) { }
private IEdmStructuralProperty GetColorFlagsProp(IEdmModel model) { }
```

Also instead of many methods for getting different IEdmTypes, I have created one method `private T GetIEdmType<T>(string typeName) where T : IEdmType` to replace the following many methods so that in future if you want to get IEdmType of another entity, you just need to provide typeName and the type to the `GetIEdmType<T>()` method instead of adding another method:
```cs
private IEdmEnumType GetColorFlagsType(IEdmModel model)
private IEdmEnumType GetColorType(IEdmModel model)
```